src/libsync/propagateupload.h

```diff
@@ -310,6 +314,7 @@ private slots:
 private:
   PropagateUploadEncrypted *_uploadEncryptedHelper;
   bool _uploadingEncrypted;
+  QPair<SyncFileItem::Status, QString> _status = { SyncFileItem::NoStatus, QString() };
```

Comment:
I'd roll a trivial struct instead of using QPair, this makes things more readable at the call site when you have properly named fields rather than `.first` and `.second`.

Can be as simple as:
```
struct UploadStatus {
     SyncFileItem::Status status = SyncFileItem::NoStatus;
    QString message;
}
UploadStatus _status;
```

Might need to shuffle the names a bit because `_status.status` looks weird but you got the idea.